Various: Reduce SERIAL usage

orhankislal opened the pull request:
Since SERIAL columns do not guarantee a sequence of numbers with no gaps, this commit removes its usage from assoc_rules and load_keras_model.
The issue was unearthed by a change in the GPDB code setting the cache for sequences to 20 and causing the numbers to jump unexpectedly. While we can set the cache to 1 to get the previous behavior, it is better to remove the dependency.

<!--  

Thanks for sending a pull request!  Here are some tips for you:
1. Refer to this link for contribution guidelines https://cwiki.apache.org/confluence/display/MADLIB/Contribution+Guidelines
2. Please Provide the Module Name, a JIRA Number and a short description about your changes.
-->

- [ ] Add the module name, JIRA# to PR/commit and description.
- [ ] Add tests for the change.

orhankislal committed:
Various: Reduce SERIAL usage

Since SERIAL columns do not guarantee a sequence of numbers with no
gaps, this commit removes its usage from load_keras_model.
The issue was unearthed by a change in the GPDB code setting the cache
for sequences to 20 and causing the numbers to jump unexpectedly.
A similar issue is found in assoc_rules. There we wanted to avoid using
row_number so we set the cache to 1 to get the previous behavior.

khannaekta commented on src/ports/postgres/modules/assoc_rules/assoc_rules.py_in:
```diff
@@ -380,29 +380,33 @@ def assoc_rules(madlib_schema, support, confidence, tid_col,
+
+        sql = """
+            INSERT INTO {assoc_loop_aux}(id, set_list, support, tids)
+            SELECT row_number() OVER (), *
```
I understand the reason for adding `row_number()` here but the only concern I have is that it would generate a plan that will gather data on the coordinator which may have performance slow down with growing data.
we can probably rewrite this query to add rownumbers without using this window function(similar to how we do it for DL's minibatch preprocessor commit: 85eba2968b5651b6242a398df569b1f3f6412703) but that would make this too complicated. Would like to know if using serial datatype and setting the cache to 1 seems better over the performance.

orhankislal committed:
Build: Add Ubuntu flag for PyXB installation

PyXB uses distutils which is being phased out. The ubuntu images fail to
install it the old way so this commit adds a flag to identify the OS as
Ubuntu during cmake and use pip install as needed.

khannaekta reviewed (approved):
Thanks @orhankislal! LGTM
